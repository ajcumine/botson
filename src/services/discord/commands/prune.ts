import { Message } from 'discord.js';
import { Command } from './';

export const execute = (message: Message, args: string[]) => {
  const amount = parseInt(args[0], 10) + 1;
  if (isNaN(amount)) {
    return message.reply("that doesn't seem to be a valid number.");
  }

  if (amount <= 1 || amount > 100) {
    return message.reply('you need to input a number between 1 and 99.');
  }
  message.channel.bulkDelete(amount, true);
};

const prune: Command = {
  execute,
  description: 'Deletes a suppiled number of messages in the current channel',
  guildOnly: false,
  argsRequired: true,
  usage: '<number>',
};

export default prune;
