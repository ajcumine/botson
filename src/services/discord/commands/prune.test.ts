import prune, { execute } from './prune';

describe('!prune command', () => {
  it('handles !prune commands', () => {
    expect(prune).toMatchSnapshot();
  });

  describe('execute', () => {
    it('deletes the number of messages provided +1 for the trigger command', () => {
      const bulkDeleteSpy = jest.fn();
      const mockMessage = {
        channel: {
          bulkDelete: bulkDeleteSpy,
        },
      };
      const args = ['3'];

      execute(mockMessage, args);
      expect(bulkDeleteSpy).toHaveBeenCalledWith(4, true);
    });

    it('does not delete if a number is not provided', () => {
      const bulkDeleteSpy = jest.fn();
      const replySpy = jest.fn();
      const mockMessage = {
        channel: {
          bulkDelete: bulkDeleteSpy,
        },
        reply: replySpy,
      };
      const args = ['not-a-number'];

      execute(mockMessage, args);
      expect(bulkDeleteSpy).not.toHaveBeenCalled();
      expect(replySpy).toHaveBeenCalledWith(
        "that doesn't seem to be a valid number.",
      );
    });

    it('does not delete if the number provided is not in the required range', () => {
      const bulkDeleteSpy = jest.fn();
      const replySpy = jest.fn();
      const mockMessage = {
        channel: {
          bulkDelete: bulkDeleteSpy,
        },
        reply: replySpy,
      };
      const args = ['1000000'];

      execute(mockMessage, args);
      expect(bulkDeleteSpy).not.toHaveBeenCalled();
      expect(replySpy).toHaveBeenCalledWith(
        'you need to input a number between 1 and 99.',
      );
    });
  });
});
