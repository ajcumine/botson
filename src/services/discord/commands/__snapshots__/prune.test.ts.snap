// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`!prune command handles !prune commands 1`] = `
Object {
  "argsRequired": true,
  "description": "Deletes a suppiled number of messages in the current channel",
  "execute": [Function],
  "guildOnly": false,
  "usage": "<number>",
}
`;
